packages/react/src/components/ListBox/ListBoxMenuItem.tsx

```diff
@@ -12,22 +12,28 @@ import React, {
   useEffect,
   useRef,
   useState,
+  type RefObject,
 } from 'react';
 import PropTypes from 'prop-types';
 import { usePrefix } from '../../internal/usePrefix';
 import { ForwardRefReturn, ReactAttr } from '../../types/common';
 
-function useIsTruncated(ref) {
+const useIsTruncated = <T extends HTMLElement>(
+  ref: RefObject<T>,
+  deps: any[] = []
+) => {
   const [isTruncated, setIsTruncated] = useState(false);
 
   useEffect(() => {
     const element = ref.current;
-    const { offsetWidth, scrollWidth } = element;
-    setIsTruncated(offsetWidth < scrollWidth);
-  }, [ref, setIsTruncated]);
+
+    if (element) {
+      setIsTruncated(element.offsetWidth < element.scrollWidth);
+    }
+  }, [ref, ...deps]);
```

Comment:
At first glance I thought this was an issue because changing the value of the ref won't trigger a re-render, but [turns out that's not the case](https://www.epicreact.dev/why-you-shouldnt-put-refs-in-a-dependency-array#:~:text=What%20about%20a%20custom%20hook%20that%20accepts%20a%20ref). TIL!